pyccel/parser/semantic.py

```diff
@@ -1547,17 +1547,22 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign, *
                                     self._current_fst_node.col_offset))
 
                         else:
-                            # TODO [YG, 04.11.2020] If we could be sure that the
-                            # array was not created in an if-then-else block, we
-                            # would use status='allocated' instead.
+                            previous_allocations = var.get_direct_user_nodes(lambda p: isinstance(p, Allocate))
```

Comment:
Are we sure that we never get an empty list here?

Reply:
If we reach here then the variable already exists, which means it must have been allocated. If we get an empty list then it is because we are missing an `Allocate`. This may change however when functions can return arrays

Reply:
I see. Then I think we should raise a nice meaningful error if for some reason `previous_allocations` is empty. This might well save us an headache later.

Reply:
Done here: e2d6b82